ulib/FStar.BigOps.fsti

```diff
@@ -0,0 +1,224 @@
+(*
+   Copyright 2008-2018 Microsoft Research
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*)
+module FStar.BigOps
+
+(* This library provides propositional connectives over finite sets
+   expressed as lists, aka "big operators", in analogy with LaTeX
+   usage for \bigand, \bigor, etc.
+
+   The library is designed with a dual usage in mind:
+
+     1. Normalization: When applied to a list literal, we want
+       `big_and f [a;b;c]` to implicilty reduce to `f a /\ f b /\ f c`
+
+     2. Symbolic manipulation: We provide lemmas of the form
+
+        ```big_and f l <==> forall x. L.memP x l ==> f x```
+
+        In this latter form, partially computing `big_and` as a fold
+        over a list is cumbersome for proof. So, we provide variants
+        `big_and'` etc., that do not reduce implicitly.
```

Comment:
Yes, I think so. If you define your predicate using big_and and you want to prove some property about it (aside from the lemmas that the library already gives you), it will sometimes be more convenient to prove it in terms of of big_and'